### PR TITLE
 Support deleting point of a feature

### PR DESCRIPTION
**Pull request by xintongxia**
delete-demo

### CODE REVIEW

**Comment by xintongxia** [comment]
@yunyu

File: modules/react-map-gl-draw/src/editor.js [hunk 1]
```diff
@@ -138,6 +144,36 @@ export default class Editor extends PureComponent<EditorProps, EditorState> {
   _events: any;
   _context: ?MapContext;
 
+  _onKeydown = (evt: MjolnirEvent) => {
+    const { features } = this.state;
+    if (
+      features &&
+      (evt.key === 'Backspace' || evt.key === 'Delete') &&
```
Comment (macrigiuseppe) [inline modules/react-map-gl-draw/src/editor.js]: i would recommend to user event.which https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/which in order to detect the key.
backspace: 8
delete: 46

I would create constants for those values

File: modules/react-map-gl-draw/src/editor.js [hunk 2]
```diff
@@ -689,7 +720,7 @@ export default class Editor extends PureComponent<EditorProps, EditorState> {
     switch (index) {
       case hoveredVertexIndex:
         return RENDER_STATE.HOVERED;
-      case -1:
+      case 'uncommitted':
```
Comment (macrigiuseppe) [inline modules/react-map-gl-draw/src/editor.js]: create a constant for this value and you can also export to library consumers

File: modules/react-map-gl-draw/src/editor.js [hunk 3]
```diff
@@ -65,6 +67,24 @@ const defaultProps = {
   onSelect: () => {}
 };
 
+const defaultEditingState = {
+  selectedFeatureId: -1,
```
Comment (macrigiuseppe) [inline modules/react-map-gl-draw/src/editor.js]: it looks like you are using __-1__ in several places, i would create a constant

File: modules/react-map-gl-draw/src/editor.js [hunk 4]
```diff
@@ -295,7 +319,8 @@ export default class Editor extends PureComponent<EditorProps, EditorState> {
       panend: evt => this._onEvent(this._onPan, evt, false)
     };
 
-    this._context.eventManager.on(this._events, ref);
+    eventManager.on(this._events, ref);
+    eventManager.on('keydown', evt => this._onEvent(this._onKeydown, evt, false));
```
Comment (macrigiuseppe) [inline modules/react-map-gl-draw/src/editor.js]: why didn't you bind __keydown__ event as part of __this._events__ object?

**Comment by supersonicclay** [comment]
In the past, we made a design choice that converting geometry types was unexpected (with the exception of going Polygon<=>MultiPolygon). So, in the `modify` edit mode of nebula, we will shrink a polygon down to a triangle, but no further. And when removing a point of a triangular hole, we remove the hole. And when removing a point of a triangle of a MultiPolygon, we'll remove the triangle.

See this for a demo of how it works in nebula.
delete-point

**Comment by JSanchezIO** [comment]
Are there any updates on this?

**Pull request closed without merging**